include/html_dumper.h

```diff
@@ -33,7 +33,7 @@ namespace kiwix
 class HTMLDumper : public LibraryDumper
 {
  public:
-  HTMLDumper(const Library* library, const NameMapper* NameMapper);
+  HTMLDumper(ConstLibraryPtr library, std::shared_ptr<const NameMapper> NameMapper);
```

Comment:
IMHO, `HTMLDumper` and `OPDSDumper` could be replaced with free functions - they were implemented as classes for reasons unrelated to the goals and principles of object oriented design. Now, having them as classes enables creating the dumper at one place and using it later but do we really need that functionality? Isn't it better to require that they are used as if they were free functions (i.e use and discard immediately after construction while ensuring that the `Library` and `NameMapper` objects stay unchanged during that time)?